libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -737,28 +746,62 @@ bool ContractCompiler::visit(ForStatement const& _forStatement)
 		_forStatement.loopExpression()->accept(*this);
 
 	m_context.appendJumpTo(loopStart);
+
+	endVisitLoop(&_forStatement);
+
 	m_context << loopEnd;
 
 	m_continueTags.pop_back();
 	m_breakTags.pop_back();
 
+	// For the case where no break is executed:
+	// loop initialization variables have to be freed
+	popBlockScopedVariables(&_forStatement);
+
 	checker.check();
 	return false;
 }
 
 bool ContractCompiler::visit(Continue const& _continueStatement)
 {
-	CompilerContext::LocationSetter locationSetter(m_context, _continueStatement);
-	if (!m_continueTags.empty())
-		m_context.appendJumpTo(m_continueTags.back());
-	return false;
+	return visitBreakContinue(&_continueStatement);
 }
 
 bool ContractCompiler::visit(Break const& _breakStatement)
 {
-	CompilerContext::LocationSetter locationSetter(m_context, _breakStatement);
-	if (!m_breakTags.empty())
+	return visitBreakContinue(&_breakStatement);
+}
+
+unsigned ContractCompiler::loopVariablesSize()
```

Comment:
Naming: what about `stackSizeOfCurrentLoopVariables`?

Reply:
Yea, I agree the naming isn't very precise, I just usually run away from long names :p